File: tests/patterns/java/equivalence_constant_propagation_class_fields.java
class A {
  private String f1;
  private static String f2;
  private String f3 = "abc";
  private String f4;
  String f5;
  private String f6;
  static { f2 = "abc"; }
  A() { f1 = "abc"; f4 = "abc"; f5 = "abc"; this.f6 = "abc"; }
  void other() { f4 = "zzz"; }
  void m() {
    // ERROR: match
    sink(1, f1);
    // ERROR: match
    sink(2, f2);
    // ERROR: match
    sink(3, f3);
    sink(4, f4);
    sink(5, f5);
    // ERROR: match
    sink(6, this.f6);
  }
}

File: tests/patterns/ts/equivalence_constant_propagation_class_fields.ts
class A {
  private f1: string;
  private static f2: string;
  private f3 = "abc";
  private f4: string;
  f5: string;
  private readonly f7: string;
  f8 = "abc";
  constructor(private p: string = "abc") { this.f1 = "abc"; this.f4 = "abc"; this.f5 = "abc"; this.f7 = "abc"; }
  other() { this.f4 = "zzz"; }
  m() {
    // ERROR: match
    const loc = "abc"; sink(0, loc);
    sink(1, this.f1);
    sink(2, A.f2);
    sink(3, this.f3);
    sink(4, this.f4);
    sink(5, this.f5);
    // ERROR: match
    sink(7, this.f7);
    sink(8, this.f8);
    sink(11, this.p);
  }
}
